lib/AFS/CellCC/Config.pm

```diff
@@ -79,12 +79,18 @@ my @directives = (
     { key => 'db/ro/user', },
     { key => 'db/ro/pass', },
 
+    { key => 'db/ro/timeout', type => 'HASH', default => {}, },
+    { key => qr:^db/ro/timeout/[^/]+$:, },
+
     { key => 'db/rw/dsn',
       default => sub { config_get('db/ro/dsn', {conf => $_[0]}) },
     },
     { key => 'db/rw/user', },
     { key => 'db/rw/pass', },
 
+    { key => 'db/rw/timeout', type => 'HASH', default => {}, },
```

Comment:
this should default to the 'ro' value, like we do with the dsn a few lines up. I wouldn't worry about trying to "merge" the sub-values together or whatever; just if the hash isn't specified at all, we should default to the 'ro' values.

Reply:
Ack.